tests/test_client.py

```diff
@@ -0,0 +1,188 @@
+        self.minio2.create_bucket(Bucket="test-delete-objects")
+        content = "TEST_DELETE_OBJECTS"
+        self.botwo_client.put_object(Bucket="test-delete-objects", Key="test/1.txt", Body=content)  # route to minio2
+        self.botwo_client.put_object(Bucket="test-delete-objects", Key="test/2.txt", Body=content)
+
+        res = self.botwo_client.delete_objects(Bucket="delete-objects1", Delete={
+            'Objects': [
+                {
+                    'Key': '1.txt',
+                },
+                {
+                    'Key': '2.txt',
+                },
+            ]
+        })
+        self.assertEqual(res["Deleted"][0]["Key"], "test/1.txt")
+        self.assertEqual(res["Deleted"][1]["Key"], "test/2.txt")
+
+    def test_paginator(self):
+        self.minio2.create_bucket(Bucket="test-paginator")
+        content = "TEST_PAGINATOR"
+        self.botwo_client.put_object(Bucket="test-paginator", Key="dev/test/1.txt", Body=content)  # route to minio2
+        self.botwo_client.put_object(Bucket="test-paginator", Key="dev/test/2.txt", Body=content)
+        self.botwo_client.put_object(Bucket="test-paginator", Key="1.txt", Body=content)
+        self.botwo_client.put_object(Bucket="test-paginator", Key="test/3.txt", Body=content)
+
+        self.assertEqual(self.botwo_client.can_paginate(operation_name='list_objects'), True)
+
+        # Create a reusable Paginator
+        paginator = self.botwo_client.get_paginator(operation_name='list_objects')
+
+        # Create a PageIterator from the Paginator
+        page_iterator = paginator.paginate(Bucket='pagi', Prefix="test")
+
+        # Compare the iterators
+        for page in page_iterator:
+            self.assertEqual(len(page['Contents']), 2)
+
+    def test_list_objects(self):
+        self.minio2.create_bucket(Bucket="test-list-objects")
+        content = "TEST_LIST_OBJECTS"
+
+        self.botwo_client.put_object(Bucket="test-list-objects", Key="dev/test/1.txt", Body=content)  # route to minio2
+        self.botwo_client.put_object(Bucket="test-list-objects", Key="dev/test/2.txt", Body=content)
+        self.botwo_client.put_object(Bucket="test-list-objects", Key="1.txt", Body=content)
+        self.botwo_client.put_object(Bucket="test-list-objects", Key="test/3.txt", Body=content)
+
+        objects = self.botwo_client.list_objects(Bucket="list", Prefix="test")
+
+        self.assertEqual(len(objects['Contents']), 2)
+
+    def test_meta(self):
+        self.assertEqual(self.botwo_client.meta, self.minio2.meta)
+
+        self.minio2.create_bucket(Bucket="test-meta")
+        event_system = self.botwo_client.meta.events
+
+        def add_my_bucket(params, **kwargs):
+            if 'Bucket' not in params:
+                params['Bucket'] = 'test-meta'
+
+        event_system.register('provide-client-params.s3.GetBucketAcl', add_my_bucket)
+        self.botwo_client.get_bucket_acl()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.docker1.kill()
```

Comment:
Can we `stop` instead?  I prefer to `docker stop`, the container might need a bit of notice to get all of its logs out, particularly if the process failed to start.

Reply:
Done